hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/dn/checksum/TestContainerCommandReconciliation.java

```diff
@@ -557,22 +515,15 @@ public void testDataChecksumReportedAtSCM() throws Exception {
       db.getStore().flushDB();
     }
 
-    // TODO: Use On-demand container scanner to build the new container merkle tree. (HDDS-10374)
-    Files.deleteIfExists(getContainerChecksumFile(container.getContainerData()).toPath());
-    kvHandler.createContainerMerkleTreeFromMetadata(container);
+    datanodeStateMachine.getContainer().getContainerSet().scanContainer(containerID);
```

Comment:
```suggestion
   datanodeStateMachine.getContainer().getContainerSet().scanContainerWithoutGap(containerID);
```
Need to change this in the whole test suite.